sys/include/saul_reg.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_saul_reg SAUL registry
+ * @ingroup     sys
+ * @brief       Global sensor/actuator registry for SAUL devices
+ *
+ * @{
+ *
+ * @file
+ * @brief       SAUL registry interface definition
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef SAUL_REG_H
+#define SAUL_REG_H
+
+#include <stdint.h>
+
+#include "saul.h"
+#include "phydat.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   SAUL registry entry
+ */
+typedef struct saul_reg_t {
+    struct saul_reg_t *next;        /**< pointer to the next device */
+    void *dev;                      /**< pointer to the device descriptor */
+    const char *name;               /**< string identifier for the device */
+    saul_driver_t const *driver;    /**< the devices read callback */
+} saul_reg_t;
+
+/**
+ * @brief   Additional data to collect for each entry
+ */
+typedef struct {
+    const char *name;           /**< string identifier for a device */
+} saul_reg_info_t;
```

Comment:
At some point we probably may decide to either get rid of this type and use a string directly or to let `saul_reg_t` point to this, dependent if there will be more information (fields) that is added to `sauL_reg_info_t`